Allow the user to enter package tracking details. Mark II: Storage

ctarda opened the pull request:
This is the second of a few PRs which will add support for allowing users to enter package tracking details manually, with the help of the Shipment Tracking extension in WC.

This PR contains changes to the Storage layer, adding two coredata entities. This PR depends on #768 , so I will open it as a draft PR. Feel free to ignore until #768 is merged.

Ref: #689

## Testing

- Checkout branch and build. Check that there are no errors in the console (i.e. coredata migration errors)
- Run the tests

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

bummytime commented on Storage/Storage/Model/ShipmentTrackingProviderGroup+CoreDataProperties.swift:
```diff
@@ -0,0 +1,38 @@
+//
+//  Created by Cesar Tardaguila on 14/3/2019.
+//  Copyright © 2019 Automattic. All rights reserved.
+//
```
Style nit → remove the header comment here ☝️

ctarda replied:
🤦‍♂️ Argh! Sorry about that

ctarda commented:
Thanks @bummytime and @mindgraffiti 

I added the siteID to both the `ShipmentTrackingProvider` and `ShipmentTrackingProviderGroup` (since we have to create custom tracking providers, I though it would make sense to make that entity dependent on the site id as well)

I also added the helpers to load all `ShipmentTrackingProvider` and `ShipmentTrackingProviderGroup`. I am not sure we would need to load individual entities so I left those out for now, but I am happy to add them if you folks think it would be necessary
